contracts/Colony.sol

```diff
@@ -125,11 +127,36 @@ contract Colony is ColonyStorage {
   function getDomainCount() public view returns (uint256) {
     return domainCount;
   }
-
   function setFunctionReviewers(bytes4 _sig, uint8 _firstReviewer, uint8 _secondReviewer)
   private
   {
     uint8[2] memory _reviewers = [_firstReviewer, _secondReviewer];
     reviewers[_sig] = _reviewers;
   }
+
+  modifier verifyKey(bytes key) {
+    uint256 colonyAddress;
+    uint256 skillid;
+    uint256 userAddress;
+    assembly {
+        colonyAddress := mload(add(key,32))
+        skillid := mload(add(key,52)) // Colony address was 20 bytes long, so add 20 bytes
+        userAddress := mload(add(key,84)) // Skillid was 32 bytes long, so add 32 bytes
+    }
+    colonyAddress >>= 96;
```

Comment:
Couldn't think of a reason to shift here, can you help me understand? I am guessing it is to do with moving the local variable to a place in memory where it won't be disturbed?

Reply:
`mload` loads 32 bytes, starting from the slot passed as its argument. The colony address (and user address) in the key are only 20 bytes long, so without shifting there is some garbage at the end of the addresses; in the case of the `colonyAddress` it is the start of the `skillid`, and in the case of the `userAddress`... I don't know! Shifting by 96 bits (`=(32-20)*8`) shifts the garbage out, and puts the address in the rightmost part of the storage slot (which is where it's expected to be).